src/code/z_parameter.c

```diff
@@ -65,125 +65,22 @@ Input sPostmanTimerInput[4];
-    { SCENE_GORONSHOP, RESTRICTIONS_INDOORS },
-    { SCENE_DEKU_KING, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 0, 0, 3, 0, 0, 0) },
-    { SCENE_LAST_GORON, RESTRICTIONS_MOON },
-    { SCENE_24KEMONOMITI, RESTRICTIONS_NONE },
-    { SCENE_F01_B, RESTRICTIONS_NO_DOUBLE_TIME },
-    { SCENE_F01C, RESTRICTIONS_NO_DOUBLE_TIME },
-    { SCENE_BOTI, RESTRICTIONS_NONE },
-    { SCENE_HAKUGIN_BS, RESTRICTIONS_NONE },
-    { SCENE_20SICHITAI, RESTRICTIONS_NONE },
-    { SCENE_21MITURINMAE, RESTRICTIONS_NONE },
-    { SCENE_LAST_ZORA, RESTRICTIONS_MOON },
-    { SCENE_11GORONNOSATO2, RESTRICTIONS_NONE },
-    { SCENE_SEA, RESTRICTIONS_NONE },
-    { SCENE_35TAKI, RESTRICTIONS_NONE },
-    { SCENE_REDEAD, RESTRICTIONS_NONE },
-    { SCENE_BANDROOM, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 0, 0, 3, 0, 0, 0) },
-    { SCENE_11GORONNOSATO, RESTRICTIONS_NONE },
-    { SCENE_GORON_HAKA, RESTRICTIONS_NONE },
-    { SCENE_SECOM, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 3, 0, 3, 0, 0, 0) },
-    { SCENE_10YUKIYAMANOMURA, RESTRICTIONS_NONE },
-    { SCENE_TOUGITES, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 3, 3, 3, 3, 0, 0) },
-    { SCENE_DANPEI, RESTRICTIONS_NO_DOUBLE_TIME },
-    { SCENE_IKANAMAE, RESTRICTIONS_NONE },
-    { SCENE_DOUJOU, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 0, 0, 3, 0, 0, 1) },
-    { SCENE_MUSICHOUSE, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 0, 0, 3, 0, 0, 0) },
-    { SCENE_IKNINSIDE, RESTRICTIONS_SET(0, 0, 0, 0, 0, 3, 3, 3, 3, 0, 0, 0) },
-    { SCENE_MAP_SHOP, RESTRICTIONS_INDOORS },
-    { SCENE_F40, RESTRICTIONS_NONE },
-    { SCENE_F41, RESTRICTIONS_NONE },
-    { SCENE_10YUKIYAMANOMURA2, RESTRICTIONS_NONE },
-    { SCENE_14YUKIDAMANOMITI, RESTRICTIONS_NONE },
-    { SCENE_12HAKUGINMAE, RESTRICTIONS_NONE },
-    { SCENE_17SETUGEN, RESTRICTIONS_NONE },
-    { SCENE_17SETUGEN2, RESTRICTIONS_NONE },
-    { SCENE_SEA_BS, RESTRICTIONS_NONE },
-    { SCENE_RANDOM, RESTRICTIONS_NONE },
-    { SCENE_YADOYA, RESTRICTIONS_INDOORS },
-    { SCENE_KONPEKI_ENT, RESTRICTIONS_NONE },
-    { SCENE_INSIDETOWER, RESTRICTIONS_SET(0, 0, 0, 0, 3, 3, 3, 3, 3, 0, 0, 0) },
-    { SCENE_26SARUNOMORI, RESTRICTIONS_NO_DOUBLE_TIME },
-    { SCENE_LOST_WOODS, RESTRICTIONS_NONE },
-    { SCENE_LAST_LINK, RESTRICTIONS_MOON },
-    { SCENE_SOUGEN, RESTRICTIONS_MOON },
-    { SCENE_BOMYA, RESTRICTIONS_INDOORS },
-    { SCENE_KYOJINNOMA, RESTRICTIONS_NONE },
-    { SCENE_KOEPONARACE, RESTRICTIONS_NO_DOUBLE_TIME },
-    { SCENE_GORONRACE, RESTRICTIONS_NONE },
-    { SCENE_TOWN, RESTRICTIONS_NONE },
-    { SCENE_ICHIBA, RESTRICTIONS_NONE },
-    { SCENE_BACKTOWN, RESTRICTIONS_NONE },
-    { SCENE_CLOCKTOWER, RESTRICTIONS_NONE },
-    { SCENE_ALLEY, RESTRICTIONS_NONE },
+#include "tables/scene_table.h"
     // { RESTRICTIONS_TABLE_END, RESTRICTIONS_NONE }, // See note below
 };
 //! @note: in `Interface_SetSceneRestrictions`, `RESTRICTIONS_TABLE_END` act as a terminating value to
 // stop looping through the array. If a scene is missing, then this will cause issues.
```

Comment:
I suppose this issue is less important now that restrictions are forced to be part of the scene definition now

Reply:
Should I remove this comment?

Reply:
I'm not sure. I think it could stay since the while-loop processing these restrictions was designed around `RESTRICTIONS_TABLE_END`

Reply:
A little tangent regarding this and does not need to be addressed in this PR. This looks the the same pattern everywhere with paths, where the loop terminates when it finds a scene with a `(u8)-1` (0xFF) id. Perhaps instead of `RESTRICTIONS_TABLE_END 0xFF` we have `RESTRICTIONS_SCENE_NONE (u8)-1`?